### ISSUE TITLE
RFE: ICMPv6 parser

### ISSUE THREAD

**Issue by daviessm** [bestouff/pktparse-rs#15]
I'd like to be able to parse ICMPv6 packets.

**Comment by bestouff** [comment]
Would you go as far as submitting some code ?

**Comment by daviessm** [comment]
Yes, I'll pick it up myself if/when I get time but I'm equally happy for someone else to get there first!

**Comment by bestouff** [comment]
Great ! Please submit a PR when you're ready, we'll make sure to review it.

**Comment by daviessm** [comment]
I've started work on this at https://github.com/daviessm/pktparse-rs/tree/icmpv6 - currently it only parses the "type" and "code" fields (and isn't fully tested). ICMPv6 is a big beast and I doubt I'll ever be able to code the whole lot of possible options. At what point would you like me to submit a PR?

**Comment by bestouff** [comment]
Well, as long as you produce something useful and correct, even if not complete I have no difficulties accepting a PR.  
Could you add some tests with real-life ICMPv6 packets ?

**Comment by daviessm** [comment]
I've added tests for echo request and reply. Can you see whether this is the style you'd like it to be written in?

Also, the echo payload data isn't parsed (because it can't be) and is returned as-is. It seems that the ICMPv4 parser does the same thing.
